Add static asset build and s3-deploy deploy make task

metasoarous opened the pull request.

crkrenn committed:
added static build/deploy to Makefile

ballPointPenguin committed:
small improvements to asset Makefile targets

metasoarous committed:
Merge branch 'edge' into ck/build_and_deploy_makefile

ballPointPenguin commented:
I'm inclined to place the `build/` folder inside of `file-server/` so that it can optionally be used by (non-dockerized) file-server directly in some cases. e.g. `cd file-server && npm start`

metasoarous committed:
Merge branch 'edge' into ck/build_and_deploy_makefile

metasoarous committed:
remove upload-web-assets make command

This command does not function as intended, since some assets are compressed, and the `aws s3` command is not actually designed to do anything with the `*.headersJson` files which specify the metadata needed for s3 static site hosting to serve these assets correctly.

metasoarous committed:
Rewrite the bin/deploy-static-assets to use aws-cli

This gets around authentication issues with IAM Identity Center users, which require SSO, and aren't well supported by all AWS SDKs.

metasoarous commented:
> I'm inclined to place the `build/` folder inside of `file-server/` so that it can optionally be used by (non-dockerized) file-server directly in some cases. e.g. `cd file-server && npm start`

@ballPointPenguin Interesting idea... Wonder if it wouldn't be better to just have this be an environment variable to configure this? I'm not sure why/when someone would want to do this though, so let's table for now, and fire up another issue if it feels right.

metasoarous committed:
remove S3_BUCKET from make config

This isn't needed, since we're reverting back to the old script for publishing to s3
